osu.Game/Graphics/UserInterface/ShearedSearchTextBox.cs

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Input;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class ShearedSearchTextBox : CompositeDrawable, IHasCurrentValue<string>
+    {
+        private const float icon_container_width = 50;
+        private const float corner_radius = 10;
+        private const float height = 42;
+        private readonly Vector2 shear = new Vector2(0.2f, 0);
+
+        public FocusedTextBox TextBox;
```

Comment:
found this when I'm trying to change the current `TestSceneDrawableLoungeRoom`'s textbox with this new search textbox. since this new component doesn't directly inherit `FocusedTextBox` like the existing `SearchTextBox`, we don't have direct access to the `FocusedTextBox`'s members without exposing it to public.

example (can be included in this PR if you want):
```diff
diff --git a/osu.Game.Tests/Visual/Multiplayer/TestSceneDrawableLoungeRoom.cs b/osu.Game.Tests/Visual/Multiplayer/TestSceneDrawableLoungeRoom.cs
index ea895a23d2..70685650d0 100644
--- a/osu.Game.Tests/Visual/Multiplayer/TestSceneDrawableLoungeRoom.cs
+++ b/osu.Game.Tests/Visual/Multiplayer/TestSceneDrawableLoungeRoom.cs
@@ -30,7 +30,7 @@ public class TestSceneDrawableLoungeRoom : OsuManualInputManagerTestScene
         protected readonly OverlayColourProvider ColourProvider = new OverlayColourProvider(OverlayColourScheme.Pink);

         private DrawableLoungeRoom drawableRoom;
-        private SearchTextBox searchTextBox;
+        private ShearedSearchTextBox searchTextBox;

         private readonly ManualResetEventSlim allowResponseCallback = new ManualResetEventSlim();

@@ -63,9 +63,8 @@ public void SetUpSteps()
                     RelativeSizeAxes = Axes.Both,
                     Children = new Drawable[]
                     {
-                        searchTextBox = new SearchTextBox
+                        searchTextBox = new ShearedSearchTextBox
                         {
-                            HoldFocus = true,
                             Anchor = Anchor.TopCentre,
                             Origin = Anchor.TopCentre,
                             Margin = new MarginPadding(50),
@@ -79,6 +78,8 @@ public void SetUpSteps()
                         }
                     }
                 };
+
+                searchTextBox.TextBox.HoldFocus = true;
             });
         }

@@ -87,7 +88,7 @@ public void TestFocusViaKeyboardCommit()
         {
             DrawableLoungeRoom.PasswordEntryPopover popover = null;

-            AddAssert("search textbox has focus", () => checkFocus(searchTextBox));
+            AddAssert("search textbox has focus", () => checkFocus(searchTextBox.TextBox));
             AddStep("click room twice", () =>
             {
                 InputManager.MoveMouseTo(drawableRoom);
@@ -114,7 +115,7 @@ public void TestFocusViaKeyboardCommit()
             AddStep("press escape", () => InputManager.Key(Key.Escape));
             AddStep("press escape", () => InputManager.Key(Key.Escape));

-            AddUntilStep("search textbox has focus", () => checkFocus(searchTextBox));
+            AddUntilStep("search textbox has focus", () => checkFocus(searchTextBox.TextBox));
         }

         [Test]
@@ -122,7 +123,7 @@ public void TestFocusViaMouseCommit()
         {
             DrawableLoungeRoom.PasswordEntryPopover popover = null;

-            AddAssert("search textbox has focus", () => checkFocus(searchTextBox));
+            AddAssert("search textbox has focus", () => checkFocus(searchTextBox.TextBox));
             AddStep("click room twice", () =>
             {
                 InputManager.MoveMouseTo(drawableRoom);
@@ -158,7 +159,7 @@ public void TestFocusViaMouseCommit()
                 InputManager.Click(MouseButton.Left);
             });

-            AddUntilStep("search textbox has focus", () => checkFocus(searchTextBox));
+            AddUntilStep("search textbox has focus", () => checkFocus(searchTextBox.TextBox));
         }

         private bool checkFocus(Drawable expected) =>
```

Reply:
Do it the other way around. Rather than expose the internal textbox outwardly, expose a matching property and proxy:

```csharp
public class ShearedSearchTextBox
{
    private SearchTextBox searchBox;
    // ...
    public bool HoldFocus
    {
         get => searchBox.HoldFocus;
         set => searchBox.HoldFocus = value;
    }
}
```

is the general idea. Gets more finicky if you initialise the textbox in BDL but the same principle applies, with a backing field and some `IsLoaded` checks.